### ISSUE TITLE
jupyter hangs when more than one notebook is open

### ISSUE THREAD

**Issue by soottikkal** [OSC/bc_osc_jupyter_spark#6]
Jupyter hangs if more than one ipynb files are open.

**Comment by nickjer** [comment]
Is this with Spark? If so it is because of the FIFO (first in, first out) nature of the standalone Spark cluster. The first notebook will get all of the Spark cluster resources, and the second notebook will be blocked until the first notebook releases those resources.

What is the use case for the issue you are experiencing? Maybe a solution would be we allow the user to choose to start either a PySpark notebook or a regular Python notebook (this notebook won't block since it doesn't request resources from the Spark cluster).

**Comment by soottikkal** [comment]
Yes, this was with Jupyter+Spark app. During the OSC workshop, some students clicked on both read-only tutorial file and its copy. This resulted in having two Jupyter sessions. I had to shut-down the read-only session to proceed with the tutorials.
